azure/scope/cluster.go

```diff
@@ -220,19 +221,22 @@ func (s *ClusterScope) RouteTableSpecs() []azure.RouteTableSpec {
 	return routetables
 }
 
-// NatGatewaySpecs returns the node nat gateway.
-func (s *ClusterScope) NatGatewaySpecs() []azure.NatGatewaySpec {
-	var natGateways []azure.NatGatewaySpec
+// NatGatewaySpecs returns the node NAT gateway.
+func (s *ClusterScope) NatGatewaySpecs() []azure.ResourceSpecGetter {
+	natGateways := []azure.ResourceSpecGetter{}
```

Comment:
```suggestion
	var natGateways []azure.ResourceSpecGetter
```

https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1812#discussion_r740309773